lib/wireprotocol/2_6_support.js

```diff
@@ -1,207 +1,140 @@
 'use strict';
 
-var copy = require('../connection/utils').copy,
-  retrieveBSON = require('../connection/utils').retrieveBSON,
-  KillCursor = require('../connection/commands').KillCursor,
-  GetMore = require('../connection/commands').GetMore,
-  Query = require('../connection/commands').Query,
-  f = require('util').format,
-  MongoError = require('../error').MongoError,
-  getReadPreference = require('./shared').getReadPreference;
-
-var BSON = retrieveBSON(),
-  Long = BSON.Long;
-
-var WireProtocol = function() {};
-
-//
-// Execute a write operation
-var executeWrite = function(pool, bson, type, opsField, ns, ops, options, callback) {
+const retrieveBSON = require('../connection/utils').retrieveBSON;
+const KillCursor = require('../connection/commands').KillCursor;
+const GetMore = require('../connection/commands').GetMore;
+const Query = require('../connection/commands').Query;
+const MongoError = require('../error').MongoError;
+const getReadPreference = require('./shared').getReadPreference;
+const applyCommonQueryOptions = require('./shared').applyCommonQueryOptions;
+const BSON = retrieveBSON();
+const Long = BSON.Long;
+
+class WireProtocol {
+  insert(pool, ns, bson, ops, options, callback) {
+    executeWrite(pool, bson, 'insert', 'documents', ns, ops, options, callback);
+  }
+
+  update(pool, ns, bson, ops, options, callback) {
+    executeWrite(pool, bson, 'update', 'updates', ns, ops, options, callback);
+  }
+
+  remove(pool, ns, bson, ops, options, callback) {
+    executeWrite(pool, bson, 'delete', 'deletes', ns, ops, options, callback);
+  }
+
+  killCursor(bson, ns, cursorState, pool, callback) {
+    const cursorId = cursorState.cursorId;
+    const killCursor = new KillCursor(bson, ns, [cursorId]);
+    const options = {
+      immediateRelease: true,
+      noResponse: true
+    };
+
+    if (typeof cursorState.session === 'object') {
+      options.session = cursorState.session;
+    }
+
+    if (pool && pool.isConnected()) {
+      try {
+        pool.write(killCursor, options, callback);
+      } catch (err) {
+        if (typeof callback === 'function') {
+          callback(err, null);
+        } else {
+          console.warn(err);
+        }
+      }
+    }
+  }
+
+  getMore(bson, ns, cursorState, batchSize, connection, options, callback) {
+    const getMore = new GetMore(bson, ns, cursorState.cursorId, { numberToReturn: batchSize });
+    function queryCallback(err, result) {
+      if (err) return callback(err);
+      const r = result.message;
+
+      // If we have a timed out query or a cursor that was killed
+      if ((r.responseFlags & (1 << 0)) !== 0) {
```

Comment:
I'm guessing that this is left over from before, but why the `1 << 0` instead of just `1`?

Reply:
Definitely left from before, but I wonder if we're missing more here. The [OP_REPLY](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#op-reply) documentation indicates the following for `responseFlags`:
```
0 corresponds to CursorNotFound. Is set when getMore is called but the cursor id is not valid at the server. Returned with zero results.
1 corresponds to QueryFailure. Is set when query failed. Results consist of one document containing an “$err” field describing the failure.
2 corresponds to ShardConfigStale. Drivers should ignore this. Only mongos will ever see this set, in which case, it needs to update config from the server.
3 corresponds to AwaitCapable. Is set when the server supports the AwaitData Query option. If it doesn’t, a client should sleep a little between getMore’s of a Tailable cursor. Mongod version 1.6 supports AwaitData and thus always sets AwaitCapable.
```

but we have them defined in [commands.js](https://github.com/mongodb-js/mongodb-core/blob/master/lib/connection/commands.js#L25-L28) as:
```
var CURSOR_NOT_FOUND = 0;
var QUERY_FAILURE = 2;
var SHARD_CONFIG_STALE = 4;
var AWAIT_CAPABLE = 8;
```

Looks like we're checking for an invalid bit here

Reply:
Okay, after much discussion and investigation I've identified a few points of interest:
  - it doesn't matter that we're defining `CURSOR_NOT_FOUND` as `0` in `commands.js`, because we only ever check the response flag in the wire protocol files, and there we check the correct value (which is `1`). 

  - half of those flags don't matter anymore, as they were used for feature detection in early versions of the server. 

Nonetheless, I will export the values and use them by name with a corrected value.